datasets/utils/continual_dataset.py

```diff
@@ -35,12 +40,83 @@ def __init__(self, args: Namespace) -> None:
             raise NotImplementedError(
                 'The dataset must be initialized with all the required fields.')
 
+        if args.n_slots or args.n_drifts or args.sequential_drifts:
+            n_tasks = self.N_TASKS
+            n_classes = self.N_CLASSES_PER_TASK * self.N_TASKS
+            self.stream_spec = StreamSpecification(n_tasks, n_classes, random_seed=args.seed,
+                                                   n_slots=args.n_slots, n_drifts=args.n_drifts, sequential_drifts=args.sequential_drifts)
+            self.stream_spec_it = iter(self.stream_spec)
+
     def get_data_loaders(self) -> Tuple[DataLoader, DataLoader]:
         """
         Creates and returns the training and test loaders for the current task.
         The current training loader and all test loaders are stored in self.
         :return: the current training and test loaders
         """
+        new_classes = list(range(self.i, self.i + self.N_CLASSES_PER_TASK))
+        train_dataset = self.get_dataset(train=True)
+        train_dataset.select_classes(new_classes)
+        test_dataset = self.get_dataset(train=False)
+        test_dataset.select_classes(new_classes)
+
+        train_loader = DataLoader(train_dataset,
+                                  batch_size=self.args.batch_size, shuffle=True, num_workers=4)
+        test_loader = DataLoader(test_dataset,
+                                 batch_size=self.args.batch_size, shuffle=False, num_workers=4)
+        self.train_loader = train_loader
+        self.test_loaders.append(test_loader)
+
+        self.i += self.N_CLASSES_PER_TASK
+        return train_loader, test_loader
+
+    def get_drifted_data_loaders(self) -> Tuple[DataLoader, DataLoader]:
+        task_spec: list[int] = next(self.stream_spec_it)
+
```

Comment:
Unused variable

Reply:
We talked about it.  The StreamSpec class should be rewritten to return current and drifted classes separately when iterating over it. I will work on that.